reader_concurrency_semaphore.cc

```diff
@@ -1460,11 +1460,8 @@ future<> reader_concurrency_semaphore::do_wait_admission(reader_permit::impl& pe
     if (admit != can_admit::yes || !_wait_list.empty()) {
         auto fut = enqueue_waiter(permit, wait_on::admission);
         if (admit == can_admit::yes && !_wait_list.empty()) {
-            // This is a contradiction: the semaphore could admit waiters yet it has waiters.
-            // Normally, the semaphore should admit waiters as soon as it can.
-            // So at any point in time, there should either be no waiters, or it
-            // shouldn't be able to admit new reads. Otherwise something went wrong.
-            maybe_dump_reader_permit_diagnostics(*this, "semaphore could admit new reads yet there are waiters", nullptr);
+            // The semaphore can admit waiters yet it has waiters. Wake the execution loop
+            // to process the waiters
```

Comment:
Let's add a mention of the fact that we can have delayed processing of resource release at play here.

Reply:
Done